src/python/pants/backend/python/goals/tailor.py

```diff
@@ -90,6 +91,30 @@ def is_entry_point(content: bytes) -> bool:
     return _entry_point_re.search(content) is not None
 
 
+@rule_helper
+async def _find_resource_py_typed_targets(
+    py_typed_files_globs: PathGlobs, all_owned_sources: AllOwnedSources
+) -> list[PutativeTarget]:
+    """Find resource targets that may be created after discovering any `py.typed` files."""
+    all_py_typed_files = await Get(Paths, PathGlobs, py_typed_files_globs)
+    unowned_py_typed_files = set(all_py_typed_files.files) - set(all_owned_sources)
+    classified_unowned_py_typed_files = {ResourceTarget: unowned_py_typed_files}
+
+    putative_targets = []
+    for tgt_type, paths in classified_unowned_py_typed_files.items():
```

Comment:
Why iterate over a singleton dict? We could get rid of the dict entirely no?

Reply:
Yes, indeed. We iterate over a dict in other tailor rules, so I kept it there for consistency, but can see now there is no reason to have it. Fixed.